### PR TITLE
VAULT-23335: Audit - Exclusion Docs (Draft)

### PR DESCRIPTION
**Pull request by peteski22**
Exclusion docs: https://vault-bhp0k8428-hashicorp.vercel.app/vault/docs/enterprise/audit/exclusion

This PR documents the Enterprise-only audit feature named 'exclusion' (see: https://github.com/hashicorp/vault-enterprise/pull/5590)

### CODE REVIEW

File: website/content/docs/enterprise/audit/exclusion.mdx [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+    "fields": [ "/request/data" ]
+  }
+]
+```
+#### Multiple exclusions
+
+This example uses multiple exclusions to exclude:
+
+* `data` from both the request and response when the mount type is `transit`.
+* `entity_id` from requests where the `/auth/client_token` in the entry matches `hmac` and then more than one other character.
```
Comment (schavis) [inline website/content/docs/enterprise/audit/exclusion.mdx:99]:
```suggestion
Use multiple JSON blocks to exclude:

* `data` from both the request and response when the mount type is `transit`.
* `entity_id` from requests where the `/auth/client_token` starts with `hmac`
  followed by at least one other character.
```

It's worth clarifying if `\"hmac.+\" only matches to strings that *start* with "hmac" or any string that contains "hmac" with at least one character after it. I'm assuming the former in the edit, but feel free to adjust the suggestion if it's the latter.

File: website/content/docs/enterprise/audit/exclusion.mdx [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+## Audit entry structure
+
+In order to accurately construct `condition` and `fields` the Vault Operator will
+have to understand the structure of the (request/response) audit entry.
+
+This information can be obtained by reverse engineering the audit logs, however it
+the following can be used to understand the structure in a more proactive way.
+
+At a high level, there are only request audit entries and response audit entries,
+each of these entries contain different objects such as `auth`, `request` and `response`.
```
Comment (schavis) [inline website/content/docs/enterprise/audit/exclusion.mdx:123]:
```suggestion
To accurately construct `condition` and `fields`, Vault operators need a solid
understanding of their audit entry structures. At a high level, there are only
**request** audit entries and **response** audit entries, but each of these
entries can contain different objects such as `auth`, `request` and `response`.

We strongly encourage operaters to review existing audit logs from a timeframe
of at least 2-4 weeks to better identify appropriate exclusion conditions and
fields.
```

File: website/content/docs/enterprise/audit/exclusion.mdx [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+  "response": <response>,
+  "time": "",
+  "type": ""
+}
+```
+
+### Auth object (`<auth>`)
+
+Complete with example data to show simple types (`string`, `bool`, `int`), other
+JSON examples which refer to `<auth>` are referring to this object.
```
Comment (schavis) [inline website/content/docs/enterprise/audit/exclusion.mdx:155]:
```suggestion
The following auth object definition includes example data with simple types
(`string`, `bool`, `int`) and used in other JSON examples that include an
`<auth>` object.
```
Style correction: avoid "this" as a pronoun

File: website/content/docs/enterprise/audit/exclusion.mdx [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+  "token_issue_time": "",
+  "token_ttl": 3600,
+  "token_type": ""
+}
+```
+
+### Request object (`<request>`)
+
+Complete with example data to show simple types (`string`, `bool`, `int`), other
+JSON examples which refer to `<request>` are referring to this object.
```
Comment (schavis) [inline website/content/docs/enterprise/audit/exclusion.mdx:208]:
```suggestion
The following request object definition includes example data with simple types
(`string`, `bool`, `int`) and used in other JSON examples that include an
`<request>` object.
```

File: website/content/docs/enterprise/audit/exclusion.mdx [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+  "replication_cluster": "",
+  "request_uri": "",
+  "wrap_ttl": 60
+}
+```
+
+### Response object (`<response>`)
+
+Complete with example data to show simple types (`string`, `bool`, `int`), other
+JSON examples which refer to `<response>` are referring to this object.
```
Comment (schavis) [inline website/content/docs/enterprise/audit/exclusion.mdx:244]:
```suggestion
The following response object definition includes example data with simple types
(`string`, `bool`, `int`) and used in other JSON examples that include an
`<response>` object.
```

File: website/content/docs/enterprise/audit/exclusion.mdx [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+    "wrapped_accessor": ""
+  },
+  "warnings": [
+    ""
+  ]
+}
+```
+## JSON schema
+
+The JSON schema for both request and response audit entries are included below.
```
Comment (schavis) [inline website/content/docs/enterprise/audit/exclusion.mdx:277]:
```suggestion
```

I don't think the section heading is helping here. I'd recommend removing it and bumping the request/response sections to `h2` so they show in the TOC

File: website/content/docs/enterprise/audit/exclusion.mdx [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+
+### Request audit entry schema
```
Comment (schavis) [inline website/content/docs/enterprise/audit/exclusion.mdx:279]:
```suggestion
## Request audit entry schema
```

File: website/content/docs/enterprise/audit/exclusion.mdx [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+
+@include 'audit/request-entry-json-schema.mdx'
+
+### Response audit entry schema
```
Comment (schavis) [inline website/content/docs/enterprise/audit/exclusion.mdx:283]:
```suggestion
## Response audit entry schema
```

File: website/content/partials/audit-options-common.mdx [hunk 9]
```diff
@@ -1,6 +1,11 @@
 - `elide_list_responses` `(bool: false)` - See [Eliding list response
 bodies](/vault/docs/audit#eliding-list-response-bodies).
 
+- `exclude` `(string: "")` - <EnterpriseAlert inline="true" /> Allows a number of
+exclusions to be provided which, when their condition is matched, will remove the
+specified fields from the audit entry being written. See the [exclusion](/vault/docs/enterprise/audit/exclusion)
+section of the auditing overview for more information.
```
Comment (schavis) [inline website/content/partials/audit-options-common.mdx:7]:
```suggestion
- `exclude` `(string: "")` - <EnterpriseAlert inline="true" /> Defines a set of 
  rules such that, when the condition is matched, Vault removes the specified
  fields from the audit entry before writing to the audit log. Refer to the
  [exclusion](/vault/docs/enterprise/audit/exclusion) section of the auditing
  overview for more information.
```
Style correction: write in active voice

**Review by schavis — CHANGES_REQUESTED** [review]

File: website/content/docs/enterprise/audit/exclusion.mdx [hunk 10]
```diff
@@ -0,0 +1,285 @@
...
+  Learn about the behavior and syntax for excluding audit data in Vault Enterprise.
+---
+
+# Exclusion syntax for audit results
+
+@include 'alerts/enterprise-only.mdx'
+
+As of Vault 1.18.0, you can enable audit devices with an `exclude` option to exclude
+particular fields in an audit entry, written to a particular audit log and fine-tune
+your auditing process.
```
[the remark this answers is not in the document]
Comment (peteski22) [reply website/content/docs/enterprise/audit/exclusion.mdx:14]: @schavis would this be acceptable? 

```suggestion
particular fields in an audit entry that is written to a particular audit log, and fine-tune
your auditing process.
```

**Review by schavis — APPROVED** [review]

File: website/content/docs/enterprise/audit/exclusion.mdx [hunk 11]
```diff
@@ -0,0 +1,285 @@
...
+  **Always** test your audit configuration in a non-production environment
+  before deploying exclusions to production. Read the
+  [Vault security model](/vault/docs/internals/security) and
+  [filtering overview](/vault/docs/concepts/filtering) to familiarize yourself
+  with Vault auditing and filtering basics before enabling audit devices that use
+  exclusions.
+
+</Warning>
+
+Once you enable an audit device with exclusions, every audit entry Vault sends to
```
Comment (vishalnayak) [inline website/content/docs/enterprise/audit/exclusion.mdx:30]: `every audit entry Vault` might take paraphrasing a little bit for easy reading.

**Review by vishalnayak — APPROVED** [review]

**Pull request merged**